File: Leet Code/1678 - Goal Parser.py
class Solution:
    def interpret(self, command: str) -> str:
        buf = ''
        ret_str = ''
        d = dict(zip(('G', '()', '(al)'), ('G', 'o', 'al')))
        for i in range(len(command)):
            buf += command[i]
            if buf in d:
                ret_str += d[buf]
                buf = ''

        return ret_str

File: Leet Code/1678_Goal_Parser.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution 
{
    public:
        string interpret(string command) 
        {
            string buff,ret_str;
            int i;
            for (i=0;i<command.length();i++)
            {
                buff.push_back(command[i]);
                switch(buff)
                {
                    case 'G':
                        ret_str.push_back('G');
                        
                }
            }

        }
};